app/models/spree/order.rb

```diff
@@ -611,6 +612,11 @@ def address_from_distributor
 
     private
 
+    def set_unused_address_fields
+      ship_address.company = 'Company' if ship_address.present?
+      bill_address.company = 'Company' if bill_address.present?
+    end
```

Comment:
Same code as above. Should this be on the address or in a module?

Reply:
I'm not sure I understand what you mean by that?

Reply:
You copied the `set_unused_address_fields` method into many models. If you want to change something about the method, you need to change it in lots of places. So it's better to define that in one place and include it in all these models. An example would be:

https://github.com/openfoodfoundation/openfoodnetwork/blob/02a10ede152a21f7bac8dcc386b102903a2b5aa5/app/models/concerns/balance.rb#L7-L13

But this is a really small method and I'm wondering if the code belongs to the model. I think that the real problem is that the `Spree::Address` model doesn't really fit out purposes. Or we use it in different ways.

One idea we could experiment with is to set the `company` field in a `before_save` action of normal address records. And for the business address, you could create a new model `BusinessAddress` which has different logic but stores data in the same database table. What do you think about that? Maybe discuss that with someone in your timezone?